devserver/devserver/devserver.go

```diff
@@ -166,40 +166,49 @@ func CreateFileHandler(servingPath, manifest string, pkgs []string, base string)
 	return indexOnNotFoundHandler
 }
 
-// chainedDir implements http.FileSystem by looking in the list of dirs one after each other.
-type chainedDir []http.Dir
+// dirHttpFileSystem implements http.FileSystem by looking in the list of dirs one after each other.
+type dirHttpFileSystem []string
+
+func (packageNames dirHttpFileSystem) Open(name string) (http.File, error) {
+	for _, packageName := range packageNames {
+		filePackageName := filepath.Join(packageName, name)
+		realFilePath, err := bazel.Runfile(filePackageName)
 
-func (chain chainedDir) Open(name string) (http.File, error) {
-	for _, dir := range chain {
-		f, err := dir.Open(name)
-		if os.IsNotExist(err) {
-			continue
-		}
 		if err != nil {
-			return nil, err
+			// In case the runfile could not be found, we also need to check that the requested
+			// path does not refer to a directory containing an "index.html" file. This can
+			// happen if Bazel runs without runfile symlinks, where only files can be resolved
+			// from the manifest. In that case we dirty check if there is a "index.html" file.
+			realFilePath, err = bazel.Runfile(filepath.Join(filePackageName, "index.html"))
```

Comment:
Do you think we can reuse the `FileSystem` implementations from `concatjs` here?

Reply:
Not sure. Specifically because only `conatjs` uses the `FileSystem` because it should remain Bazel independent. This file is Bazel-aware and uses the `bazel.Runfile` logic directly.

I could see the point though. Let me know what you prefer. I don't feel too strongly about it.

Reply:
We can keep it the way it is

Reply:
Sounds good.